### ISSUE TITLE
`prettier/prettier` reports, but cannot be fixed automatically

### ISSUE THREAD

**Issue by JounQin** [mdx-js/eslint-mdx#491]
### Initial checklist

- [X] I read the [support docs](https://mdxjs.com/community/support/)
- [X] I read the [contributing guide](https://mdxjs.com/community/contribute/)
- [X] I agree to follow the [code of conduct](https://github.com/mdx-js/.github/blob/HEAD/code-of-conduct.md)
- [X] I searched [issues](https://github.com/search?q=user%3Amdx-js&type=Issues) and couldn’t find anything (or linked relevant results below)

### Affected packages and versions

eslint-plugin-mdx

### Steps to reproduce

```mdx
import     {     Baz   } from     './Fixture'

export  const   foo    = {
  hi:     `Fudge ${Baz.displayName || 'Baz'}`,
  authors: [
     'fred',
           'sally'
    ]
}

export const hello = 'hello'

# Hello,    world!

I'm an awesome   paragraph.

{ /* I'm a comment */ }

<Foo bg="red">
    <Bar>hi                                            </Bar>

  {  hello       }

  {     /* another comment */}
</Foo>

```
test codeblock
```

```js
module.exports = 'test'
```

```sh
npm i -g foo
```

| Test | Table |
| :--- | :---- |
| Col1 | Col2  |

export   default     ({children   }) => < div>{    children}</div>
```

### Expected behavior

Format with `eslint-plugin-prettier` same as `prettier` cli itself

### Actual behavior

Incorrect formatting

### Runtime

Node v20

### Package manager

yarn v1

### OS

macOS

### Build and bundle tools

Rollup
